platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java

```diff
@@ -341,14 +339,15 @@ HttpUnsuccessfulResponseHandler getHttpUnsuccessfulResponseHandler(Message messa
         String method = (String) message.get(Message.HTTP_REQUEST_METHOD);
         HttpContent content = null;
         if (!isRedirectable(method)) {
-          ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-          message.setContent(OutputStream.class, byteArrayOutputStream);
-          BodyWriter bodyWriter = new BodyWriter();
-          bodyWriter.handleMessage(message);
-          content =
-              new InputStreamContent(
-                  (String) message.get(Message.CONTENT_TYPE),
-                  new ByteArrayInputStream(byteArrayOutputStream.toByteArray()));
+          try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
+            message.setContent(OutputStream.class, tfbos);
+            BodyWriter bodyWriter = new BodyWriter();
+            bodyWriter.handleMessage(message);
+            ByteSource byteSource = tfbos.asByteSource();
+            content =
+                new InputStreamContent(method, byteSource.openStream())
```

Comment:
❓ Should we be passing in the HTTP Method as the "type" parameter instead of `(String) message.get(Message.CONTENT_TYPE)`

Reply:
Yes!